pkg/security/admission/admission.go

```diff
@@ -177,11 +196,13 @@ func assignSecurityContext(provider scc.SecurityContextConstraintsProvider, pod
 	}
 
 	if len(errs) > 0 {
+		// ensure psc is not mutated if there are errors
+		pod.Spec.SecurityContext = originalPSC
 		return errs
 	}
 
 	// if we've reached this code then we've generated and validated an SC for every container in the
-	// pod so let's apply what we generated
+	// pod so let's apply what we generated.  Note: the psc is already applied.
 	for i, sc := range generatedSCs {
 		pod.Spec.Containers[i].SecurityContext = sc
```

Comment:
Think you worked this one out yourself below.  Making the naming change to the loop above so it is more obvious that we aren't changing the original container.